packages/strapi-plugin-content-manager/admin/src/components/DynamicZone/index.js

```diff
@@ -78,6 +87,42 @@ const DynamicZone = ({
     [layout, name]
   );
 
+  const dynamicComponentCategories = useMemo(() => {
```

Comment:
Here we will use lodash `groupBy` instead to simplify the code a bit:

```
const dynamicComponentCategories = useMemo(() => {
  const componentsWithInfos = dynamicZoneAvailableComponents.map(componentUid => {
    const {
      category,
      schema: { info },
    } = getDynamicComponent(componentUid);

    return { componentUid, category, info };
  });

  return groupBy(componentsWithInfos, 'category');
}, [dynamicZoneAvailableComponents, getDynamicComponent]);
```

This will return an object like the following:

```
{
  category1: [ { componentUid, category: category1: info }, ....],
}
```

Reply:
@soupette done in 15570e056d4c6c19e4615bd214dd63fbff7df881